app/code/Magento/CmsGraphQl/Model/Resolver/Page.php

```diff
@@ -44,35 +45,61 @@ public function resolve(
         array $value = null,
         array $args = null
     ) {
-        $pageId = $this->getPageId($args);
-        $pageData = $this->getPageData($pageId);
+        if (!isset($args['id']) && !isset($args['identifier'])) {
+            throw new GraphQlInputException(__('"Page id/identifier should be specified'));
+        }
+
+        if (isset($args['id'])) {
+            $pageData = $this->getPageDataById($this->getPageId($args));
+        } elseif (isset($args['identifier'])) {
+            $pageData = $this->getPageDataByIdentifier($this->getPageIdentifier($args));
+        }
 
         return $pageData;
     }
 
     /**
      * @param array $args
      * @return int
-     * @throws GraphQlInputException
      */
     private function getPageId(array $args): int
     {
-        if (!isset($args['id'])) {
-            throw new GraphQlInputException(__('"Page id should be specified'));
-        }
+        return isset($args['id']) ? (int)$args['id'] : 0;
+    }
 
-        return (int)$args['id'];
+    /**
+     * @param array $args
+     * @return string
+     */
+    private function getPageIdentifier(array $args): string
+    {
+        return isset($args['identifier']) ? (string)$args['identifier'] : '';
     }
 
     /**
      * @param int $pageId
      * @return array
      * @throws GraphQlNoSuchEntityException
      */
-    private function getPageData(int $pageId): array
+    private function getPageDataById(int $pageId): array
+    {
+        try {
+            $pageData = $this->pageDataProvider->getDataByPageId($pageId);
+        } catch (NoSuchEntityException $e) {
```

Comment:
Could you please move this `try/catch` statement into `\Magento\CmsGraphQl\Model\Resolver\Page::resolve` and use it for both - `getDataByPageId` and `getPageDataByIdentifier` instead of copy&paste it?
It seems that this method (`\Magento\CmsGraphQl\Model\Resolver\Page::getPageDataById`) will not be needed at all in case you will move `try/catch` statement.

Reply:
**done**